p8e-api-webservice/src/main/kotlin/io/provenance/p8e/webservice/domain/ApiAffiliate.kt

```diff
@@ -43,9 +43,10 @@ fun AffiliateShareRecord.toApi(): ApiAffiliateShare =
         created
     )
 
-data class RegisterAffiliateKey(val signingPrivateKey: String?, val encryptionPrivateKey: String?, val useSigningKeyForEncryption: Boolean, val indexName: String, val alias: String?) {
+data class RegisterAffiliateKey(val signingPrivateKey: String?, val encryptionPrivateKey: String?, val authKeyPair: String?, val useSigningKeyForEncryption: Boolean, val indexName: String, val alias: String?) {
     val signingKeyPair: KeyPair = signingPrivateKey.toOrGenerateKeyPair()
     val encryptionKeyPair: KeyPair = signingKeyPair.takeIf { useSigningKeyForEncryption }.or { encryptionPrivateKey.toOrGenerateKeyPair() }
+    val authPublicKey: PublicKey = authKeyPair.toOrGenerateKeyPair().public
```

Comment:
within this class, I think we might need to actually keep the whole key pair, as the response will include the private key (only time we send the private key back, so the user has it in the case that it was generated)

Reply:
Got it, wasn't clear of how this would look from the UI perspective and if was strict on which key should be used/set. I'll make the change.